File: java/ta-core/src/main/java/com/mh/ta/core/config/driver/browser/Firefox.java

package com.mh.ta.core.config.driver.browser;

import static org.openqa.selenium.remote.DesiredCapabilities.firefox;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxProfile;
import org.openqa.selenium.remote.DesiredCapabilities;

/**
 * @author minhhoang
 *
 */
class Firefox implements SeleniumDriver {
	private FirefoxProfile profile;
	private DesiredCapabilities capabilities = firefox();

	public WebDriver createDriver() {
		DesiredCapabilities cap = DesiredCapabilities.firefox();
		cap.setCapability(FirefoxDriver.PROFILE, getProfile());
		return new FirefoxDriver(cap.merge(getCapabilities()));		
	}

	private FirefoxProfile getProfile() {
		if (profile == null) {
			profile = new FirefoxProfile();
			profile.setAcceptUntrustedCertificates(true);
			profile.setAlwaysLoadNoFocusLib(true);
			profile.setAssumeUntrustedCertificateIssuer(true);
		}
		return profile;
	}

	private DesiredCapabilities getCapabilities() {
		if (capabilities == null)
			capabilities = firefox();
		return capabilities;
	}

	public void setDriverOptions(Object options) {
		this.profile = (FirefoxProfile) options;
	}

	public void setCapabilities(Object capabilities) {
		this.capabilities = (DesiredCapabilities) capabilities;
	}
}

File: python/Core/Config/SeleniumDriver/Firefox.py
from selenium import webdriver
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
from selenium.webdriver.firefox.options import Options

from .Driver import SeleniumDriver


class Firefox(SeleniumDriver):
    def __init__(self):
        super().__init__()

    def create_driver(self):
        if self._driver is None:
            self._driver = webdriver.Firefox(capabilities=None, firefox_options=None)

    def set_driver_desired_capabilities(self, desired_capabilities):
        if isinstance(desired_capabilities, DesiredCapabilities):
            self.capabilities = desired_capabilities

    def set_driver_options(self, options):
        if isinstance(options, Options):
            self.options = options

    __module__ = "firefox"
